test/scheduler/utils.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <gtest/gtest.h>
+
+#include <ctime>
+#include <type_traits>
+
+#include "../../source/event.hpp"
+#include "../../source/scheduler.hpp"
+
+class TestScheduler : public testing::Test {
+public:
+    void TearDown() override{
+        sim::Scheduler::get_instance().clear();
+    };
+    void SetUp() override{};
+};
+
+namespace sim {
```

Comment:
I think it would better to use other namespace:
```suggestion
namespace test {
```